CMakeLists.txt

```diff
@@ -173,7 +173,7 @@ elseif(CMAKE_COMPILER_IS_GNUCXX)
         "${CMAKE_CXX_FLAGS} -faligned-new -Werror=implicit-fallthrough=2")
     endif()
     set(CMAKE_CXX_FLAGS
-      "${CMAKE_CXX_FLAGS} -Wunused-result -Werror=unused-result -Wunused-parameter -Werror=unused-parameter")
+      "${CMAKE_CXX_FLAGS} -Wunused-result -Werror=unused-result -Wno-unused-parameter")
```

Comment:
Why is this needed? Would prefer to keep this warning on.

Reply:
Without this change, I got compile errors at grpc cpp generator code:

```
/tmp/flatbuffers/grpc/src/compiler/cpp_generator.cc: In function ‘void grpc_cpp_generator::PrintHeaderClientMethodCallbackInterfacesEnd(grpc_generator::Printer*, std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >*)’:
/tmp/flatbuffers/grpc/src/compiler/cpp_generator.cc:606:43: error: unused parameter ‘vars’ [-Werror=unused-parameter]
     std::map<grpc::string, grpc::string>* vars) {
                                           ^~~~
/tmp/flatbuffers/grpc/src/compiler/cpp_generator.cc: In function ‘void grpc_cpp_generator::PrintHeaderClientMethodCallbackStart(grpc_generator::Printer*, std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >*)’:
/tmp/flatbuffers/grpc/src/compiler/cpp_generator.cc:622:43: error: unused parameter ‘vars’ [-Werror=unused-parameter]
     std::map<grpc::string, grpc::string>* vars) {
                                           ^~~~
/tmp/flatbuffers/grpc/src/compiler/cpp_generator.cc: In function ‘void grpc_cpp_generator::PrintHeaderClientMethodCallbackEnd(grpc_generator::Printer*, std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >*)’:
/tmp/flatbuffers/grpc/src/compiler/cpp_generator.cc:669:43: error: unused parameter ‘vars’ [-Werror=unused-parameter]
     std::map<grpc::string, grpc::string>* vars) {
                                           ^~~~
```

Reply:
There already is:
```
 if(FLATBUFFERS_BUILD_GRPCTEST)
   if(CMAKE_COMPILER_IS_GNUCXX)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-parameter -Wno-shadow")
   endif()
...
```
lets keep it specific to GRPC.